custom_components/divoom_pixoo/sensor.py

```diff
@@ -73,81 +75,96 @@ async def async_will_remove_from_hass(self):
-                    try:
-                        rendered_text = str(text_template.async_render())
-                    except TemplateError as e:
-                        _LOGGER.error("Template render error: %s", e)
-                        rendered_text = "Template Error"
-
-                    if text['font'] == "FONT_PICO_8":
-                        pixoo.draw_text(rendered_text, tuple(text['position']), tuple(text['font_color']), FONT_PICO_8)
-                    if text['font'] == "FONT_GICKO":
-                        pixoo.draw_text(rendered_text.upper(), tuple(text['position']), tuple(text['font_color']),
-                                        FONT_GICKO)
-                    if text['font'] == "FIVE_PIX":
-                        pixoo.draw_text(rendered_text.upper(), tuple(text['position']), tuple(text['font_color']), FIVE_PIX)
-
-            if "PV" in self.page:
-                solar(pixoo, self.hass, self.page, FONT_PICO_8, FONT_GICKO)
-
-            if "Fuel" in self.page:
-                _LOGGER.info(f"Fuel = {self.page}")
-                _LOGGER.info(f"Fuel = {self.page['Fuel']}")
-                fuel(pixoo, self.hass, self.page, FONT_PICO_8, FONT_GICKO, FIVE_PIX, ELEVEN_PIX)
-
-            if "channel" not in self.page and "clockId" not in self.page:
-                pixoo.push()
+        if len(self._pages) == 0:
+            return
 
-        page_displayed = False
         initial_index = self._current_page_index
-
-        while not page_displayed:
+        is_enabled = None
+        while not is_enabled:
             self.page = self._pages[self._current_page_index]
             self.schedule_update_ha_state()
 
             try:
-                is_enabled = str(Template(self.page['enable'], self.hass).async_render()) if 'enable' in self.page else "true"
+                is_enabled = str(Template(str(self.page.get('enabled', 'true')), self.hass).async_render())
                 is_enabled = is_enabled.lower() in ['true', 'yes', '1', 'on']
             except TemplateError as e:
                 _LOGGER.error(f"Error rendering enable template: {e}")
                 is_enabled = False
 
             if is_enabled:
-                _LOGGER.info(f"Yes it is True!")
-                await self.hass.async_add_executor_job(update)
+                await self.hass.async_add_executor_job(self._render_page, self.page)
                 self._current_page_index = (self._current_page_index + 1) % len(self._pages)
-                page_displayed = True
             else:
-                _LOGGER.info(f"No it is False!")
                 self._current_page_index = (self._current_page_index + 1) % len(self._pages)
                 if self._current_page_index == initial_index:
+                    _LOGGER.info("All pages disabled. Not updating.")
                     break
 
+    def _render_page(self, page):
+        pixoo = self._pixoo
+        pixoo.clear()
+
+        if page['page_type'].lower() == "channel":
```

Comment:
I believe the `pages` are just a dictionary at this point, but I'm wondering if it might work to have a `Page`  (or whatever name you like) abstraction  that contains the data from the config plus a `render` method that takes the `pixoo` as an argument. Then you could have subtypes that inherit from `Page` and implement `render` differently depending on the type of page they are. (Or maybe composition would be better ... maybe `Page` could have a different renderer depending on the type of the page).

Either way, just a thought and _totally_ get that you're still working on it, this giant `if/else` just keeps on calling out to me for attention :-)

Reply:
I'm a Ruby and Java programmer primarily, so I completely understand where you're coming from. Certainly I think it would improve the readability of this code, but also, I see you're still working on fully implementing the config restructure. I have found it to be a good idea to get a thing working first and then refactor for that sort of thing as a separate step. :-) Meanwhile, I'll can play around with this and see if I can come up with a concrete suggestion for how to implement that. Does that sound good?

Reply:
Sounds perfect!